File: practice/mathematics/algebra/pairwise_sum_and_divide/pairwise_sum_and_divide.cpp
#include <iostream>

using namespace std;

int main()
{
    int t, n;
    unsigned long long a, cnt1, cnt2, s;
    
    cin >> t;
    while (t--)
    {
        cin >> n;
        cnt1 = cnt2 = 0;
        for (int cnt = 0; cnt < n; ++cnt)
        {
            cin >> a;
            if (a == 1)
            {
                cnt1++;
            }
            else if (a == 2)
            {
                cnt2++;
            }
        }
        
        s = cnt1 * (n - 1) + cnt2 * (cnt2 - 1) / 2;
        cout << s << endl;
    }
    
    return 0;
}

File: practice/mathematics/algebra/pairwise_sum_and_divide/pairwise_sum_and_divide.py
#!/usr/bin/env python

t = int(raw_input().strip())
for _ in xrange(t):
    n = int(raw_input().strip())
    a = map(int, raw_input().strip().split())
    cnt_1 = 0
    cnt_2 = 0
    for POS in xrange(n):
        if a[POS] == 1:
            cnt_1 += 1
        elif a[POS] == 2:
            cnt_2 += 1
    s = cnt_1 * (n - 1) + cnt_2 * (cnt_2 - 1) / 2
    print s

